tmt/steps/execute/internal.py

```diff
@@ -172,8 +172,11 @@ def log(key, value=None, color=None, shift=1, level=1):
         test.real_duration = self.test_duration(start, end)
         duration = click.style(test.real_duration, fg='cyan')
         shift = 1 if self.opt('verbose') < 2 else 2
+        test._result = self.check(test)
```

Comment:
The `Test` class does not have an instance attribute `_result` as far as I can see and this is the first time it is being set. I don't really like "abusing" protected class attributes for sharing data. I think you could return the result from `execute()` since it currently does not return anything. I think that would make sense, test execution returns its result. Also, doesn't `self.check()` require files to be pulled from the guest? It seems to read log files.

But I like the idea behind this change, it looks nice!

Reply:
Hm, good point about fetching.